org.ekstep.assetbrowser-1.3/editor/assetbrowserapp.js

```diff
@@ -1185,6 +1197,44 @@ angular.module('assetbrowserapp').controller('browsercontroller', ['$scope', '$i
             icon: 'fa fa-warning'
         });
     }
+    ctrl.applyVideoFilter = function() {
+        var value = ctrl.filterType;
+        ctrl.offset =0
+        var searchText = (ctrl.query === "") ? undefined : ctrl.query;
+        var selectedValue, contentType, createdBy;
+        contentType = new Array('Asset');
+        if(ctrl.tabSelected == 'my'){
```

Comment:
This is duplicate code for "my" & "all"

Reply:
tabselected all and my all is different

Reply:
if (value == 'youtube') {
	selectedValue = ctrl.videoMimeTypes.youtube
} else if(value =='video') {
	selectedValue = ctrl.videoMimeTypes.video
}else{
	selectedValue = ctrl.videoMimeTypes.all
}
            
			
if(ctrl.tabSelected == 'all') {
	if((value =='all') || (value =='youtube')) {
		createdBy = undefined;
		contentType.push('Resource');
	}
} else if(ctrl.tabSelected == 'my') {
	if((value =='all') || (value =='youtube')) {
		createdBy = ctrl.createdBy;
	}
}